src/TwigComponent/README.md

```diff
@@ -338,6 +338,44 @@ If you're using [Live Components](https://github.com/symfony/ux-live-component),
 then there _are_ some guidelines related to how the re-rendering of parent
 and child components works. Read [Live Embedded Components](https://github.com/symfony/ux-live-component#embedded-components).
 
+## Self-Rendering Components
+
+To take full control of the rendering of your component, have it implement
+`Symfony\UX\TwigComponent\SelfRenderingInterface`. This allows you create
+_dynamic templates_ based on component data:
+
+```php
+// src/Components/AlertComponent.php
+// ...
+
+use Symfony\UX\TwigComponent\SelfRenderingInterface;
+
+#[AsTwigComponent('alert')]
+class AlertComponent implements SelfRenderingInterface
+{
+    public string $message;
+    public string $type = 'success';
+
+    public function mount(bool $isSuccess = true)
+    {
+        $this->type = $isSuccess ? 'success' : 'danger';
+    }
+
+    public function render(\Twig\Environment $twig) : string
+    {
+        $template = 'components/alert-'.$this->type.'.html.twig';
+        $template = $this->getLoader()->exists($template) ? $template : 'components/alert-default.html.twig';
```

Comment:
Maybe this line is not completely necessary for this example and we can assume both templates `'components/alert-'.$this->type.'.html.twig'` already exist in our side.

Reply:
I was trying to demonstrate a complex example where we'd need this feature but I think this isn't it.